src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -50,7 +52,7 @@ private static bool UseLoggerMessageDefine(LoggerMethod lm)
                     int count = 0;
                     foreach (string t in lm.TemplateList)
                     {
-                        if (!t.Equals(lm.TemplateParameters[count].Name, StringComparison.OrdinalIgnoreCase))
+                        if (!t.Equals(lm.TemplateParameters[count++].Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
this diff allows using LoggerMessage.Define on more than 1 arguments

Reply:
As `TemplateList` is a 
https://github.com/dotnet/runtime/blob/2223babdd49118787c675e04aff711f936a10b26/src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Parser.cs#L637
maybe use a `for`-loop?

`count` isn't used outside of that loop.